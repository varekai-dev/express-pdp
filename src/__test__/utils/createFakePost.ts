import { faker } from '@faker-js/faker'
import request from 'supertest'
import { app } from '../../server'

export async function createFakePost(accessToken: string) {
	const fakePost = {
		title: faker.lorem.words(3),
		content: faker.lorem.words(3),
		file: {
			originalname: 'test.jpg',
			mimetype: 'image/jpeg',
			buffer: Buffer.from(new ArrayBuffer(1024)), // Mock 1KB JPEG image
		},
	}

	const response = await request(app)
		.post('/api/v1/posts')
		.set('Authorization', `Bearer ${accessToken}`)
		.field('title', fakePost.title)
		.field('content', fakePost.content)
		.attach('file', fakePost.file.buffer, {
			filename: fakePost.file.originalname,
			contentType: fakePost.file.mimetype,
		})
console.log('response',response)
	return {
		requestData: {
			title: fakePost.title,
			content: fakePost.content,
		},
		responseData: response.body,
	}
}
