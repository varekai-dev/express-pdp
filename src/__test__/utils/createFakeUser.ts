import { faker } from '@faker-js/faker'
import request from 'supertest'
import { app } from '../../server'

export async function createFakeUser() {
	try {
		const fakeUser = {
			username: faker.internet.userName().substring(0, 8),
			email: faker.internet.email(),
			password: faker.internet.password({
				length: 8,
			}),
		}
		const userResponse = await request(app)
			.post('/api/v1/auth/register')
			.send(fakeUser)

		if (userResponse) {
			return {
				...userResponse?.body,
				password: fakeUser.password,
			}
		}
	} catch (error) {
		console.error('createFakeUser', error)
	}
}
