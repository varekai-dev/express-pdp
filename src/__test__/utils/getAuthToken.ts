import request from 'supertest'
import { app } from '../../server'
import { fakeUser } from '../data/fakeUser'

export async function getAuthToken() {
	await request(app).post('/api/v1/auth/register').send(fakeUser)
	const response = await request(app).post('/api/v1/auth/login').send({
		email: fakeUser.email,
		password: fakeUser.password,
	})

	return response.body.accessToken
}
