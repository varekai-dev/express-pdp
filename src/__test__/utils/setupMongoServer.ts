import { MongoMemoryServer } from 'mongodb-memory-server'
import mongoose from 'mongoose'

let mongoServer: MongoMemoryServer

export async function setupMongoServer() {
	try {
		mongoServer = new MongoMemoryServer({
			instance: {
				dbName: 'jest',
			},
		})
		await mongoServer.start()
		const mongoUri = mongoServer.getUri()
		await mongoose.connect(mongoUri)
	} catch (e) {
		console.error('setupMongoServer', e)
	}
}

export async function teardownMongoServer() {
	try {
		await mongoose.connection.dropDatabase()
		await mongoose.connection.close()
		await mongoServer.stop()
	} catch (e) {
		console.error('teardownMongoServer', e)
	}
}

export async function clearMongoServerDb() {
	const collections = mongoose.connection.collections
	for (const key in collections) {
		const collection = collections[key]
		await collection.deleteMany({})
	}
}
