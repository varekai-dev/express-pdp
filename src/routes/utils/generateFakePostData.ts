import { faker } from '@faker-js/faker'

export function generateFakePostData() {
	return {
		title: faker.lorem.word(),
		content: faker.lorem.word(4),
	}
}
