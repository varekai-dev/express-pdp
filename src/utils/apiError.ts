interface ApiErrorProps {
	message: string
	errorCode: number
}

export class ApiError extends Error {
	errorCode: number
	constructor({ message, errorCode }: ApiErrorProps) {
		super(message)
		this.errorCode = errorCode
	}
}
