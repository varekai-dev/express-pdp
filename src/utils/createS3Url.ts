import { config } from 'dotenv'

config()

export function createS3Url(file: Express.Multer.File) {
	return `https://${process.env.AWS_BUCKET_NAME}.s3.${process.env.AWS_REGION}.amazonaws.com/${file.originalname}`
}
