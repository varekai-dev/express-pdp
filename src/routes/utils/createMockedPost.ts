import FormData from 'form-data'
// import mockFs from 'mock-fs'
import fs from 'fs'
import { generateFakePostData } from './generateFakePostData'
import path from 'path'

// Create a mock file system
// mockFs({
// 	'path/to': {
// 		'test.jpg': Buffer.from([8, 6, 7, 5, 3, 0, 9]), // this represents binary data of a file
// 	},
// })
// Now 'path/to/test.jpg' exists and can be read
// const file = fs.createReadStream('path/to/test.jpg')

export function createMockedPost() {
	const fakePostData = generateFakePostData()
	const post = new FormData()
	post.append('title', fakePostData.title)
	post.append('content', fakePostData.content)
	const filePath = path.join(__dirname, 'img', 'image.jpeg')
	const file = fs.createReadStream(filePath)
	post.append('file', file, 'image.jpeg')
	return post
}
