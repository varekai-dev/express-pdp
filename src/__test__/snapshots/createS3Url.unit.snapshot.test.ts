import { createS3Url } from '../../utils/createS3Url'

describe('createS3Url', () => {
	it('should return the correct S3 URL', () => {
		const result = createS3Url({
			originalname: 'test.jpg',
		} as Express.Multer.File)

		expect(result).toMatchSnapshot()
	})
})
