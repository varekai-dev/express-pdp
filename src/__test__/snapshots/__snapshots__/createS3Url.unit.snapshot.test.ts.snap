// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createS3Url should return the correct S3 URL 1`] = `"https://express-pdp.s3.eu-central-1.amazonaws.com/test.jpg"`;
